internal/snmp/wrapper.go

```diff
@@ -0,0 +1,173 @@
+package snmp
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/soniah/gosnmp"
+)
+
+// GosnmpWrapper wraps a *gosnmp.GoSNMP object so we can use it as a snmpConnection.
+type GosnmpWrapper struct {
+	*gosnmp.GoSNMP
```

Comment:
(optional) Not a huge issue, but if you make the wrapper wrap `gosnmp.GoSNMP` instead of `*gosnmp.GoSNMP` you won't have the struct creation for error returns. Pretty minor though.

Reply:
It seems to be this way intentionally and isn't a trivial change.  The way snmp input's connection caching was written, the wrapper itself needs to be copyable but the GoSNMP library's struct itself isn't copyable.